New bug in `4.14.1`: AttributeError: 'DoctestItem' object has no attribute 'cls'

After updating `schemathesis`

```
- Updating schemathesis (4.13.0 -> 4.14.1)
```

My tests in https://github.com/wemake-services/django-modern-rest
started to error out:

```
____________________ ERROR at setup of [doctest] dmr.response.APIError ____________________
.venv/lib/python3.13/site-packages/_pytest/runner.py:353: in from_call
    result: TResult | None = func()
                             ^^^^^^
.venv/lib/python3.13/site-packages/_pytest/runner.py:245: in <lambda>
    lambda: runtest_hook(item=item, **kwds),
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
.venv/lib/python3.13/site-packages/pluggy/_hooks.py:512: in __call__
    return self._hookexec(self.name, self._hookimpls.copy(), kwargs, firstresult)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
.venv/lib/python3.13/site-packages/pluggy/_manager.py:120: in _hookexec
    return self._inner_hookexec(hook_name, methods, kwargs, firstresult)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
.venv/lib/python3.13/site-packages/pluggy/_callers.py:167: in _multicall
    raise exception
.venv/lib/python3.13/site-packages/pluggy/_callers.py:139: in _multicall
    teardown.throw(exception)
.venv/lib/python3.13/site-packages/_pytest/logging.py:843: in pytest_runtest_setup
    yield
.venv/lib/python3.13/site-packages/pluggy/_callers.py:139: in _multicall
    teardown.throw(exception)
.venv/lib/python3.13/site-packages/_pytest/capture.py:895: in pytest_runtest_setup
    return (yield)
            ^^^^^
.venv/lib/python3.13/site-packages/pluggy/_callers.py:121: in _multicall
    res = hook_impl.function(*args)
          ^^^^^^^^^^^^^^^^^^^^^^^^^
.venv/lib/python3.13/site-packages/schemathesis/pytest/plugin.py:564: in pytest_runtest_setup
    schema = StatefulSchemaMark.get(item.cls) if item.cls is not None else None
                                                 ^^^^^^^^
E   AttributeError: 'DoctestItem' object has no attribute 'cls'
```

Repro: `poetry update && poetry run pytest dmr/decorators.py` in our repo

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -2,6 +2,10 @@
 
 ## [Unreleased](https://github.com/schemathesis/schemathesis/compare/v4.14.1...HEAD) - TBD
 
+### :bug: Fixed
+
+- `AttributeError` on `DoctestItem` when schemathesis plugin is loaded. [#3663](https://github.com/schemathesis/schemathesis/issues/3663)
+
 ## [4.14.1](https://github.com/schemathesis/schemathesis/compare/v4.14.0...v4.14.1) - 2026-03-27
 
 ### :bug: Fixed

--- a/src/schemathesis/pytest/plugin.py
+++ b/src/schemathesis/pytest/plugin.py
@@ -561,7 +561,8 @@ def _push_to_xdist_workeroutput(
 
 
 def pytest_runtest_setup(item: pytest.Item) -> None:
-    schema = StatefulSchemaMark.get(item.cls) if item.cls is not None else None
+    item_cls = getattr(item, "cls", None)
+    schema = StatefulSchemaMark.get(item_cls) if item_cls is not None else None
     if schema is not None:
         # Attach a callback to the TestCase class so the state machine can hand off
         # its recorder to the report writers once the scenario finishes.
@@ -632,8 +633,9 @@ def pytest_runtest_setup(item: pytest.Item) -> None:
 
 
 def pytest_runtest_teardown(item: pytest.Item, nextitem: pytest.Item | None) -> None:
-    if item.cls is not None and StatefulSchemaMark.is_set(item.cls):
-        StatefulCallbackMark.set(item.cls, None)
+    item_cls = getattr(item, "cls", None)
+    if item_cls is not None and StatefulSchemaMark.is_set(item_cls):
+        StatefulCallbackMark.set(item_cls, None)
         for writer in item.stash.get(_STATEFUL_WRITERS_KEY, []):
             writer.close()
         return

--- a/test/pytest/test_main.py
+++ b/test/pytest/test_main.py
@@ -1590,3 +1590,19 @@ def test(case):
     )
     result = testdir.runpytest("-v", "-s")
     result.assert_outcomes(passed=1)
+
+
+def test_doctest_items_do_not_cause_attribute_error(testdir):
+    # See GH-3663
+    testdir.makepyfile(
+        mymodule="""
+def add(a, b):
+    '''
+    >>> add(1, 2)
+    3
+    '''
+    return a + b
+"""
+    )
+    result = testdir.runpytest("--doctest-modules", "mymodule.py")
+    result.assert_outcomes(passed=1)

